AzFileDiagnostics/Windows/AzFileDiagnostics.ps1

```diff
@@ -375,6 +375,45 @@ function ValidateLmCompatibilityLevel {
 
 
     
+}
+
+##############################
+#.SYNOPSIS
+# Validate the RestrictSendingNTLMTraffic. 
+#
+#.DESCRIPTION
+# Currently just check the registry key and its value. if it is set to 0 or 1, prompt user to change it to expected value. 
+#
+#.EXAMPLE
+#An example
+#
+#.NOTES
+#General notes
+##############################
+function ValidateRestrictSendingNTLMTraffic {
+
+    $RegKeyPath = "HKLM:SYSTEM\CurrentControlSet\Control\Lsa\MSV1_0"
+    $valueName = "RestrictSendingNTLMTraffic"
+
+    $Result = Get-ItemProperty -path $RegKeyPath  -Name $valueName -ErrorAction SilentlyContinue
+
+    if ($Result -eq $null) {
+        Write-Log -level success "`n[OK]: HKLM:SYSTEM\CurrentControlSet\Control\Lsa\MSV1_0|RestrictSendingNTLMTraffic IS NOT set, by default it should be 0 (or 1)" 
+
+    }
+    else {
+
+        if ( $result.RestrictSendingNTLMTraffic -ne 2 ) {
```

Comment:
Test against 0 or 1 explicitly for the success case so that if any new value greater than 2 won't mislead us